### PR TITLE
Support for subresultant generation and features for Zp

### PR DESCRIPTION
**Pull request by Ovascos**

### CODE REVIEW

File: include/polynomial.h [hunk 1]
```diff
@@ -145,9 +160,11 @@ int lp_polynomial_sgn(const lp_polynomial_t* A, const lp_assignment_t* m);
 /** Returns the interval approximation of the polynomial value */
 void lp_polynomial_interval_value(const lp_polynomial_t* A, const lp_interval_assignment_t* m, lp_interval_t* result);
 
-/** returns the sign of the polynomial in the model */
+/** returns the value of the polynomial in the model */
 lp_value_t* lp_polynomial_evaluate(const lp_polynomial_t* A, const lp_assignment_t* m);
 
+/** returns the integer value of a polynomial. Assignment m must only assign integer values */
```
Comment (ahmed-irfan) [inline include/polynomial.h:166]: The function doesn't return anything.

File: include/polynomial.h [hunk 2]
```diff
@@ -136,6 +139,18 @@ int lp_polynomial_is_univariate_m(const lp_polynomial_t* A, const lp_assignment_
 /** Returns the univariate polynomial (if univariate, or 0 otherwise) */
 lp_upolynomial_t* lp_polynomial_to_univariate(const lp_polynomial_t* A);
 
+/**
+ * Returns the univariate polynomial of A partially evaluated with m.
+ * m must assign all variables in A (but the top variable) to an integer value.
```
Comment (ahmed-irfan) [inline include/polynomial.h:144]: what happens if m doesn't assign all variables in A?

File: include/polynomial.h (hunk 1 above)
Comment (Ovascos) [reply include/polynomial.h:166]: Uses the out parameter. I'll rephrase.

File: include/polynomial.h (hunk 2 above)
Comment (Ovascos) [reply include/polynomial.h:144]: Undefined behavior. An assertion in coefficient_evaluate_integer fails.

**Review by ahmed-irfan — APPROVED** [review]

**Pull request merged**